### PR TITLE
Add axes_2d gizmo.

### LINKED ISSUE #12222: `axes_2d` gizmo
## What problem does this solve or what need does it fill?

A follow-up item from #12211 by @mweatherley:

> Implement something similar for 2d. Honestly, I have no idea if this is desired/useful, but I could probably just implement it in this PR if that's the case.

#12211 added a gizmo for rendering the XYZ axes relative to a `Transform` (or `impl TransformPoint`). This is useful for 3D, but it'd also be useful to be able to draw just the XY axes in 2D. The 3D gizmo wouldn't look quite correct in 2D, and most other gizmo methods already have 2D variants.

## What solution would you like?

Add an `axes_2d` method that renders the XY axes.

## What alternative(s) have you considered?

Leave it up to the user, or use `axes` in 2D as well. It'd just look weird in 2D, because the arrows are 3D, and the Z arrow would probably look like a cross in the middle.

### PR DESCRIPTION
**Pull request by lambertsbennett**
# Objective

This PR addresses #12222 (Fixes #12222). Simple addition to add a 2D axes gizmo. 

## Solution

- Add a new method axes_2d which takes a transform and a case length and then draws two arrows in the XY plane. 

The only thing I'm not sure about here is taking a 3D transform as an argument. It says in the transform comments that for 2D the z-axis is used for ordering, so I figured I'd keep it that way?

---

## Changelog

- Add method axes_2d.
- Update arrow_2d to also calculate the tip length depending on arrow length as in arrow.
- Add axes_2d to examples 2d_gizmos.

### CODE REVIEW

**Comment by lambertsbennett** [comment]
I see that the build/CI are failing because the compiler says that From<Vec3> is not implemented for Vec2. This is puzzling to me because I did implement it as:

```rust
impl From<Vec3> for Vec2 {
    #[inline]
    fn from(v: Vec3) -> Self {
        (v.x, v.y)
    }
}
```
I am relatively new to rust though, so perhaps this isn't correct? Any pointers here would be great! The strange thing is that the example works just fine (```2d_gizmos```). It's just when compiling the crate that there is an error.

File: crates/bevy_gizmos/Cargo.toml [hunk 1]
```diff
@@ -14,8 +14,8 @@ webgpu = []
 
 [dependencies]
 # Bevy
-bevy_pbr = { path = "../bevy_pbr", version = "0.14.0-dev", optional = true }
 bevy_sprite = { path = "../bevy_sprite", version = "0.14.0-dev", optional = true }
+bevy_pbr = { path = "../bevy_pbr", version = "0.14.0-dev", optional = true }
```
Comment (mockersf) [inline crates/bevy_gizmos/Cargo.toml:18]: you should avoid changing this

File: examples/gizmos/2d_gizmos.rs [hunk 2]
```diff
@@ -87,6 +87,8 @@ fn draw_example_collection(
         Vec2::from_angle(sin / -10. + PI / 2.) * 50.,
         YELLOW,
     );
+
+    gizmos.axes_2d(Transform::from_translation(Vec3::new(sin, 0.0, 0.0)), 25.0)
```
Comment (mockersf) [inline examples/gizmos/2d_gizmos.rs:91]: this example is already quite crowded. this will also put it on the grid gizmo already present, making it hard to read

File: crates/bevy_gizmos/src/arrows.rs [hunk 3]
```diff
@@ -163,3 +170,36 @@ impl<'w, 's, T: GizmoConfigGroup> Gizmos<'w, 's, T> {
...
+    /// }
+    /// # bevy_ecs::system::assert_is_system(draw_axes_2d);
+    /// ```
+    pub fn axes_2d(&mut self, transform: impl TransformPoint, base_length: f32) {
+        let start = transform.transform_point(Vec3::ZERO);
+        let end_x = transform.transform_point(base_length * Vec3::X);
+        let end_y = transform.transform_point(base_length * Vec3::Y);
+
+        self.arrow_2d(start.into(), end_x.into(), RED);
+        self.arrow_2d(start.into(), end_y.into(), GREEN);
```
Comment (mockersf) [inline crates/bevy_gizmos/src/arrows.rs:203]:
```suggestion
        self.arrow_2d(start.xy(), end_x.xy(), RED);
        self.arrow_2d(start.xy(), end_y.xy(), GREEN);
```

and you'll need to import ``bevy_math::Vec3Swizzles`

File: crates/bevy_gizmos/src/arrows.rs [hunk 4]
```diff
@@ -125,7 +125,14 @@ impl<'w, 's, T: GizmoConfigGroup> Gizmos<'w, 's, T> {
...
     ) -> ArrowBuilder<'_, 'w, 's, T> {
-        self.arrow(start.extend(0.), end.extend(0.), color)
+        let length = (end - start).length();
+        ArrowBuilder {
+            gizmos: self,
+            start: start.extend(0.),
+            end: end.extend(0.),
+            color: color.into(),
+            tip_length: length / 10.,
+        }
```
Comment (mockersf) [inline crates/bevy_gizmos/src/arrows.rs:135]: Why did you change this?

Comment (lambertsbennett) [reply crates/bevy_gizmos/src/arrows.rs:135]: I changed it to be consistent with the 3d arrow drawing method, but I can remove that.

**Comment by lambertsbennett** [comment]
Ok thanks for the feedback @notmd and @mockersf - I reverted a bunch of things and now it seems to be in a better state.

File: crates/bevy_gizmos/src/arrows.rs [hunk 5]
```diff
@@ -163,3 +163,36 @@ impl<'w, 's, T: GizmoConfigGroup> Gizmos<'w, 's, T> {
         self.arrow(start, end_z, BLUE);
     }
 }
+
+impl<'w, 's, T: GizmoConfigGroup> Gizmos<'w, 's, T> {
```
Comment (pablo-lua) [inline crates/bevy_gizmos/src/arrows.rs:167]: I think we should adress both methods in the same impl if possible.

**Review by pablo-lua — APPROVED** [review]
Good change, thank you. Can you put this feature in the axes example? I think we can place it in the plane on the example? We can make the bevy logo bounce around in the floor I believe.

**Comment by lambertsbennett** [comment]
@pablo-lua I moved it into the same impl block as you suggested and added a 2D axis to the axes example. My first try at that caused there to be axes flying around in space and so I pinned them to the plane. Maybe this is not exactly illustrative of the capability of the 2D axes to transform though.

**Comment by pablo-lua** [comment]
Fair, I was thinking in placing this as a `Sprite` in reality, like, spawning a Sprite of some kind and just use the transform in it to move it around and so. We can address this in another PR though, so I think we should revert it for now and do this as a follow up.

**Comment by lambertsbennett** [comment]
@pablo-lua ok reverted the example, but otherwise I hope its all good!

**Review by alice-i-cecile — APPROVED** [review]
